homeassistant/components/namecheapdns/__init__.py

```diff
@@ -15,37 +15,50 @@
 
 DOMAIN = 'namecheapdns'
 
+CONF_HOSTS = CONF_HOST + 's'  #FIXME Put in homeassistant.const ?
+
 INTERVAL = timedelta(minutes=5)
 
 UPDATE_URL = 'https://dynamicdns.park-your-domain.com/update'
 
+DEFAULT_HOST = '@'
+
 CONFIG_SCHEMA = vol.Schema({
     DOMAIN: vol.Schema({
         vol.Required(CONF_DOMAIN): cv.string,
         vol.Required(CONF_PASSWORD): cv.string,
-        vol.Optional(CONF_HOST, default='@'): cv.string,
+        vol.Optional(CONF_HOST, default='@'):
+        cv.deprecated(replacement_key=CONF_HOSTS),
```

Comment:
This does not work this way. Here's an example pull request where we deprecated `CONF_UPDATE_INTERVAL`: https://github.com/home-assistant/home-assistant/pull/20924/files

I'm also not sure that this will work when converting from a single item to a list automatically.

Reply:
You can make it work the way you want it to by doing something like this:

```
def test_deprecated_with_replacement_key_list():
    
    def _ensure_list(key: str):
        def validator(config: {}):
            if key in config:
                config[key] = cv.ensure_list(config[key])
            return config
        return validator

    deprecated_schema = vol.All(
        vol.Schema({
            vol.Optional('host'): cv.string,
            vol.Optional('hosts'): vol.All(cv.ensure_list, [cv.string])
        }),
        cv.deprecated('host', replacement_key='hosts'),
        _ensure_list('hosts')
    )

    test_data = {'host': 'www.example.com'}
    output = deprecated_schema(test_data.copy())

    assert {'hosts': ['www.example.com']} == output
```

Reply:
Ok so I am not sure I understand all of this. I just changed the main file with that validation (avoiding an imediate breaking change).

The code you propose is a test ? Should I add it to the test file ?

Reply:
No, it's not a test, it's an example showing how to use the deprecated validator to convert from host to hosts.

Since you have used that, that avoids the immediate breaking change. You also no longer need to reference host anywhere else in the component because the deprecated validator has migrated the value correctly to hosts. That's what the test data and output in the example I gave you demonstrate.

Reply:
Thanks a lot !!!